Blockzilla/UIComponents/AutocompleteTextField.swift

```diff
@@ -1,166 +1,384 @@
+    // get the new autocompletion.
+    public func textField(_ textField: UITextField, shouldChangeCharactersIn range: NSRange, replacementString string: String) -> Bool {
+        // This happens when you begin typing overtop the old highlighted
+        // text immediately after focusing the text field. We need to trigger
+        // a `didEnterText` that looks like a `clear()` so that the SearchLoader
+        // can reset itself since it will only lookup results if the new text is
+        // longer than the previous text.
+        if lastReplacement == nil {
+            autocompleteDelegate?.autocompleteTextField(self, didEnterText: "")
         }
 
-        // Fire the delegate with the text the user typed (not including the completion).
-        autocompleteDelegate?.autocompleteTextField?(self, didTextChange: textBeforeCompletion ?? "")
+        lastReplacement = string
+        return true
     }
 
-    override open func deleteBackward() {
-        lastReplacement = nil
+    func setAutocompleteSuggestion(_ suggestion: String?) {
+        let text = self.text ?? ""
+
+        guard let suggestion = suggestion, isEditing && markedTextRange == nil else {
+            hideCursor = false
+            return
+        }
 
-        guard completionRange == nil else {
-            // If we have an active completion, delete it without deleting any user-typed characters.
-            removeCompletion()
+        let normalized = normalizeString(text)
+        guard suggestion.hasPrefix(normalized) && normalized.count < suggestion.count else {
+            hideCursor = false
             return
         }
 
-        super.deleteBackward()
+        let suggestionText = String(suggestion[suggestion.index(suggestion.startIndex, offsetBy: normalized.count)...])
+        let autocompleteText = NSMutableAttributedString(string: suggestionText)
+
+        let color = UIColor.accent.withAlphaComponent(0.4)
+        autocompleteText.addAttribute(NSAttributedString.Key.backgroundColor, value: color, range: NSRange(location: 0, length: suggestionText.count))
+
+        autocompleteTextLabel?.removeFromSuperview() // should be nil. But just in case
+        autocompleteTextLabel = createAutocompleteLabelWith(autocompleteText)
+        if let l = autocompleteTextLabel {
+            addSubview(l)
+            hideCursor = true
+            forceResetCursor()
+        }
+    }
+
+    public override func caretRect(for position: UITextPosition) -> CGRect {
+        return hideCursor ? CGRect.zero : super.caretRect(for: position)
+    }
+
+    private func createAutocompleteLabelWith(_ autocompleteText: NSAttributedString) -> UILabel {
+        let label = UILabel()
+        var frame = self.bounds
+        label.attributedText = autocompleteText
+        label.font = self.font
+        label.accessibilityIdentifier = "autocomplete"
+//        label.backgroundColor = self.backgroundColor
+        label.textColor = self.textColor
```

Comment:
```suggestion
        label.accessibilityIdentifier = "autocomplete"
        label.textColor = self.textColor
```

Reply:
wondering if we need this commented out label background

Reply:
good catch